### PR TITLE
make python udf serialization format to be binary plus tensor tables

### PR DESCRIPTION
**Pull request by zhaojuanmao**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27191 skip all rpc and dist autograd spawn tests for <PY36
* **#27136 make python udf serialization format to be binary plus tensor tables**

make python udf serialization format to be binary plus tensor tables, so that tensors can be attached to autograd graph, handled in the same way as builtin operators

Differential Revision: [D17405686](https://our.internmc.facebook.com/intern/diff/D17405686/)

### CODE REVIEW

**Review by mrshenli — APPROVED** [review]
This is the dup of #26314. The previous one hit merge conflicts during landing, but the OSS PR was already closed.

**Comment by zhaojuanmao** [comment]
failed tests are not relevant

File: torch/distributed/internal_rpc_utils.py [hunk 1]
```diff
@@ -4,30 +4,125 @@
 import copyreg
 import io
 import pickle
+import six
```
Comment (fmassa) [inline torch/distributed/internal_rpc_utils.py]: This adds a dependency on `six` which wasn't present before.

You should use instead https://github.com/pytorch/pytorch/blob/master/torch/_six.py

**Review by fmassa — COMMENTED** [review]
This PR breaks torchvision CI because it adds a hard-dependency on `six` which is not translated to the binaries https://circleci.com/gh/pytorch/vision/20778?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by zhaojuanmao** [comment]
@fmassa sorry that it broke torchvision CI, wish there is a way to catch this kind of issue before landing diffs:(

**Pull request closed without merging**